src/shared/openssl-util.c

```diff
@@ -573,6 +573,215 @@ int string_hashsum(
 
 }
 #  endif
+
+int ecdh_derive_shared_secret(
+                EVP_PKEY *private_key,
+                EVP_PKEY *public_key,
+                void **ret_shared_secret,
+                size_t *ret_shared_secret_size) {
+
+        _cleanup_(EVP_PKEY_CTX_freep) EVP_PKEY_CTX *ctx = NULL;
+        _cleanup_(erase_and_freep) void *shared_secret = NULL;
+        size_t shared_secret_size;
+        int r;
+
+        assert(private_key);
+        assert(public_key);
+        assert(ret_shared_secret);
+        assert(ret_shared_secret_size);
+
+        ctx = EVP_PKEY_CTX_new(private_key, NULL);
+        if (!ctx)
+                return log_openssl_errors("Failed to allocate pkey context");
+
+        r = EVP_PKEY_derive_init(ctx);
+        if (r != 1)
```

Comment:
given we don#t really care to keep the value around I would just shorten this to:

```c
if (EVP_PKEY_derive_init(ctx) != 1)
        …
```

Reply:
(here and everywhere)